Source/WebInspectorUI/UserInterface/Views/DataGridNode.js

```diff
@@ -674,7 +674,7 @@ WI.DataGridNode = class DataGridNode extends WI.Object
 
         if (!this.isPlaceholderNode) {
             var previousGridNode = this.traversePreviousNode(true, true);
-            insertionIndex = this.dataGrid._rows.indexOf(previousGridNode);
+            insertionIndex = this.dataGrid._rows.lastIndexOf(previousGridNode);
```

Comment:
ha! clever :)

there's probably a bunch of other spots we could do this too.  i do wonder about the cost of this in other situations where we do insert near the front of the list tho :/

maybe we could add some hints somewhere that suggest "we're close to the front" or somethin

Reply:
We could certainly optimize the null path, which would cover first-item insertion.

Reply:
I wonder if sub-calls to attach could be optimized to just have the known index. Like adding the children in the `this.expanded` case (line: 693). But I don't recall if sort order could impact things. I also don't recall if `expanded` matters much in the performance heavy cases (like Timeline data grids).